# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/url-classifier/UrlClassifierFeatureSocialTrackingAnnotation.cpp
# Commit: f2df73fda468
# Full Hash: f2df73fda46859150605f33bcdaeb49ab7037720
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2019-10-19 21:40:34
# Regressor Bug: 1588899
# File Overlap Count: 1
# Description:
#   Bug 1588899 - P1. Move classification flags related method to nsIClassifiedChannel. r=Ehsan,baku
#   
#   This is where it should have been in the first place. Those attributes belong there.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49577
# ==============================================================================

diff -r b98968eb71d3 -r f2df73fda468 netwerk/url-classifier/UrlClassifierFeatureSocialTrackingAnnotation.cpp
--- a/netwerk/url-classifier/UrlClassifierFeatureSocialTrackingAnnotation.cpp	Thu Oct 17 15:24:31 2019 +0000
+++ b/netwerk/url-classifier/UrlClassifierFeatureSocialTrackingAnnotation.cpp	Sat Oct 19 04:30:24 2019 +0000
@@ -8,6 +8,7 @@
 
 #include "mozilla/AntiTrackingCommon.h"
 #include "mozilla/net/UrlClassifierCommon.h"
+#include "nsIClassifiedChannel.h"
 #include "nsContentUtils.h"
 #include "nsNetUtil.h"
 
@@ -135,19 +136,19 @@
   static std::vector<UrlClassifierCommon::ClassificationData>
       sClassificationData = {
           {NS_LITERAL_CSTRING("social-tracking-protection-facebook-"),
-           nsIHttpChannel::ClassificationFlags::
+           nsIClassifiedChannel::ClassificationFlags::
                CLASSIFIED_SOCIALTRACKING_FACEBOOK},
           {NS_LITERAL_CSTRING("social-tracking-protection-linkedin-"),
-           nsIHttpChannel::ClassificationFlags::
+           nsIClassifiedChannel::ClassificationFlags::
                CLASSIFIED_SOCIALTRACKING_LINKEDIN},
           {NS_LITERAL_CSTRING("social-tracking-protection-twitter-"),
-           nsIHttpChannel::ClassificationFlags::
+           nsIClassifiedChannel::ClassificationFlags::
                CLASSIFIED_SOCIALTRACKING_TWITTER},
       };
 
   uint32_t flags = UrlClassifierCommon::TablesToClassificationFlags(
       aList, sClassificationData,
-      nsIHttpChannel::ClassificationFlags::CLASSIFIED_SOCIALTRACKING);
+      nsIClassifiedChannel::ClassificationFlags::CLASSIFIED_SOCIALTRACKING);
 
   UrlClassifierCommon::AnnotateChannel(
       aChannel, flags,